### PR TITLE
.github: add v1.18.x to the nightly tests

### PR DESCRIPTION
**Pull request by sam-heilbron**
# Description
Add 1.18 to the list of branches that are used for nightly tests

# Context
Now that we have an 1.18.x branch, we need to ensure we are running nightly tests against it

## Testing steps
Invoke the nightly tests using the manual trigger (todo)

Screenshot 2024-12-02 at 4 15 16 PM

Logs: https://github.com/solo-io/gloo/actions/runs/12129674969

## Notes for reviewers
-

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: .github/workflows/nightly-tests.yaml [hunk 1]
```diff
@@ -12,19 +12,19 @@ on:
   # Day of the week ([0,6] with 0=Sunday)
   schedule:
     - cron: "0 5 * * *" # every day @ 05:00 UTC, run tests against latest main
-    - cron: "0 6 * * 1" # monday    @ 06:00 UTC, run expanded tests against v1.17.x
-    - cron: "0 7 * * 1" # monday    @ 07:00 UTC, run expanded tests against v1.16.x
-    - cron: "0 8 * * 1" # monday    @ 08:00 UTC, run expanded tests against v1.15.x
+    - cron: "0 6 * * 1" # monday    @ 06:00 UTC, run expanded tests against v1.18.x
+    - cron: "0 7 * * 1" # monday    @ 06:00 UTC, run expanded tests against v1.17.x
+    - cron: "0 8 * * 1" # monday    @ 07:00 UTC, run expanded tests against v1.16.x
```
Comment (jenshu) [inline .github/workflows/nightly-tests.yaml:17]:
```suggestion
    - cron: "0 7 * * 1" # monday    @ 07:00 UTC, run expanded tests against v1.17.x
    - cron: "0 8 * * 1" # monday    @ 08:00 UTC, run expanded tests against v1.16.x
```

File: .github/workflows/nightly-tests.yaml [hunk 2]
```diff
@@ -12,19 +12,19 @@ on:
...
     inputs:
       branch:
         description: "The branch to run tests against"
         type: choice
         options:
           - main
+          - v1.18.x
           - v1.17.x
           - v1.16.x
-          - v1.15.x
```
Comment (jenshu) [inline .github/workflows/nightly-tests.yaml:27]: why don't we test on 1.15? is that still supported?

Comment (sam-heilbron) [reply .github/workflows/nightly-tests.yaml:27]: In oss previously we only tested n-2, so I wanted to maintain that. In enterprise we still have n-3 support

**Review by jenshu — APPROVED** [review]

**Review by nfuden — APPROVED** [review]

**Pull request merged**